presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SwapAdjacentWindowsBySpecifications.java

```diff
@@ -66,28 +66,71 @@ private static boolean dependsOn(WindowNode parent, WindowNode child)
 
     private static int compare(WindowNode o1, WindowNode o2)
     {
+        int comparison = comparePartitionBy(o1, o2);
+        if (comparison != 0) {
+            return comparison;
+        }
+
+        comparison = compareOrderBy(o1, o2);
+        if (comparison != 0) {
+            return comparison;
+        }
+
+        // If PartitionBy and OrderBy clauses are identical, let' establish an arbitrary order to prevent non-deterministic results of swapping WindowNodes in such a case
+        return o1.getId().toString().compareTo(o2.getId().toString());
+    }
+
+    private static int comparePartitionBy(WindowNode o1, WindowNode o2)
+    {
         Iterator<Symbol> iterator1 = o1.getPartitionBy().iterator();
         Iterator<Symbol> iterator2 = o2.getPartitionBy().iterator();
 
         while (iterator1.hasNext() && iterator2.hasNext()) {
             Symbol symbol1 = iterator1.next();
             Symbol symbol2 = iterator2.next();
 
-            int comparison = symbol1.compareTo(symbol2);
-            if (comparison != 0) {
-                return comparison;
+            int partitionByComparison = symbol1.compareTo(symbol2);
```

Comment:
I renamed the `comparison` variable to `partitionByComparison` so as to distinguish this code from the code in `compareOrderBy` for better readability.